src/plugins/explore/public/components/visualizations/style_panel/unit/unit_panel.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useMemo, useRef, useEffect, useCallback } from 'react';
+import { i18n } from '@osd/i18n';
+import { EuiFieldText, EuiPopover, EuiContextMenu, EuiFormRow, EuiButtonIcon } from '@elastic/eui';
+import { UnitsCollection, getUnitById } from './collection';
+import { StyleAccordion } from '../style_accordion';
+import './style.scss';
+
+export interface UnitPanelProps {
+  unit?: string;
+  onUnitChange: (unit: string | undefined) => void;
+}
+
+export const UnitPanel = ({ unit, onUnitChange }: UnitPanelProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const [lastPanelId, setLastPanelId] = useState(0);
+
+  const selectedUnit = getUnitById(unit);
+
+  const handleChangeUnit = useCallback(
+    (item: string | undefined) => {
+      onUnitChange(item);
+    },
+    [onUnitChange]
+  );
+
+  const inputBoxButton = (
+    <EuiFieldText
+      id="input"
+      placeholder="Select a unit"
+      value={selectedUnit?.name ?? ''}
+      onClick={() => {
+        setPopover(!isPopoverOpen);
+      }}
+      readOnly={true}
+      append={
+        unit ? (
+          <EuiButtonIcon
```

Comment:
Please add `aria-label` for `EuiButtonIcon`, otherwise it shows warning in console.